processing/src/main/java/io/druid/query/topn/TopNMapFn.java

```diff
@@ -35,6 +37,8 @@ public TopNMapFn(
   {
     this.query = query;
     this.topNAlgorithm = topNAlgorithm;
+    this.dimension = query.getDimensionSpec().getDimension();
+    this.extractionFn = query.getDimensionSpec().getExtractionFn();
```

Comment:
This is outside the scope of this PR so I'm removing it. Originally I had placed them here in order to account for only calling the methods once per query (instead of once per cursor) but since I don't have any performance tests on this and it is causing concern it is easier to just remove it.